geometry/perspective_test.cpp

```diff
@@ -143,6 +146,259 @@ TEST_F(PerspectiveTest, IsHiddenBySphere) {
   EXPECT_FALSE(perspective.IsHiddenBySphere(p4, sphere));
 }
 
+TEST_F(PerspectiveTest, VisibleSegments) {
+  // The camera is on the x-axis and looks towards the positive x.
+  Point<Displacement<World>> const camera_origin =
+      World::origin + Displacement<World>({-10 * Metre, 0 * Metre, 0 * Metre});
+  AffineMap<World, Camera, Length, OrthogonalMap> const world_to_camera_affine(
+      camera_origin,
+      Camera::origin,
+      OrthogonalMap<World, Camera>::Identity());
+  Perspective<World, Camera, Length, OrthogonalMap> perspective(
+      world_to_camera_affine,
+      /*focal=*/1 * Metre);
+
+  // The sphere is at the origin and has unit radius.
+  Sphere<Length, World> const sphere(World::origin,
+                                     /*radius=*/1 * Metre);
+
+  // The equation of the line KP in the plane x-z is:
+  //   x - 3 * sqrt(11) z + 10 = 0
+  // This is used to compute many of the expected values below.
+
+  // A segment away from the sphere with x > 0.
+  {
```

Comment:
Make those individual tests within a fixture that contains the stuff you are reusing; possibly put that in a new fixture in a new file.